Combinator/Combinator.fs

```diff
@@ -169,14 +169,26 @@ module Combinator =
     let manyTill p pEnd = manyTillWithCount 0 p pEnd
 
     let manyTill1 p pEnd = manyTillWithCount 1 p pEnd
-        
-    let manyN num (parser) =                              
-        takeTillWithCount 0 (fun i -> List.length i > num) parser    >>= fun result -> 
-            if result.Length <> num then
-                failwith ("Error, attempted to match " + num.ToString() + " but only got " + result.Length.ToString())
 
-            preturn result          
-    
+
+
+    let manyN num parser state =
+        let didFind found currentState =
```

Comment:
Seems like other partial functions, including manyN, are also not necessarily closures. So, I included the state argument. Since a closure is treated as an object, it is better to eliminate needless closures for performance reasons.